CMakeLists.txt

```diff
@@ -1,227 +1,175 @@
+option(Slicer_BUILD_EXTENSIONMANAGER_SUPPORT    "Build application with ExtensionManager support"     OFF)
+option(Slicer_BUILD_MULTIVOLUME_SUPPORT         "Build application with MultiVolume support"          OFF)
+option(Slicer_BUILD_PARAMETERSERIALIZER_SUPPORT "Build application with parameter serializer support" OFF)
+option(Slicer_USE_PYTHONQT                      "Build application with Python support"               ON)
+option(Slicer_USE_PYTHONQT_WITH_TCL             "Build application with Python Tcl adapter layer"     OFF)
+option(Slicer_USE_QtTesting                     "Build application with QtTesting support"            OFF)
+option(Slicer_USE_SimpleITK                     "Build application with SimpleITK support"            OFF)
+
+option(Slicer_BUILD_BRAINSTOOLS                 "Build application with BRAINSTools module"           OFF)
+option(Slicer_BUILD_DataStore                   "Build application with DataStore module"             OFF)
+option(Slicer_BUILD_CompareVolumes              "Build application with ChangeTrackerPy module"       OFF)
+option(Slicer_BUILD_LandmarkRegistration        "Build application with LandmarkRegistration module"  OFF)
+
+# Enable Slicer built-in modules
+set(Slicer_CLIMODULES_ENABLED
+  LabelMapSmoothing
+  ModelMaker
+  ResampleScalarVectorDWIVolume # Needed by 'CropVolume' module
   )
-set_property(GLOBAL APPEND PROPERTY ${kit}_INCLUDE_DIRS
-  ${CMAKE_CURRENT_BINARY_DIR}
+set(Slicer_QTLOADABLEMODULES_ENABLED
+  )
+set(Slicer_QTSCRIPTEDMODULES_ENABLED
   )
 
-# --------------------------------------------------------------------------
-# Sub projects
-# --------------------------------------------------------------------------
-include(ExportLibrary)
+# Disable Slicer built-in modules
+set(Slicer_CLIMODULES_DISABLED
+  )
+set(Slicer_QTLOADABLEMODULES_DISABLED
+    #Cameras
+    #Units
+    #Terminologies
+    #Colors
+    #SubjectHierarchy
+    #Annotations
+    #Markups
+    #Transforms
+    #Data
+    #DoubleArrays
+    #Models
+    #SceneViews # Needed to avoid warning when loading scene
+    #Segmentations
+    SlicerWelcome
+    #Tables
+    #Reformat
+    #ViewControllers
+    #Volumes
+    #VolumeRendering
+  )
+set(Slicer_QTSCRIPTEDMODULES_DISABLED
+  #DataProbe
+  DMRIInstall
+  Editor
+  EditorLib
+  Endoscopy
+  LabelStatistics
+  PerformanceTests
+  SampleData
+  #ScreenCapture
+  #SegmentEditor
+  #SegmentStatistics
+  SelfTests
+  SurfaceToolbox
+  VectorToScalarVolume
+  )
 
-set(${kit}_LAUNCHER_EXECUTABLE
-    ${CMAKE_CURRENT_BINARY_DIR}/../Slicer-build/${kit})
-set(${kit}_LAUNCH_COMMAND
-    ${${kit}_LAUNCHER_EXECUTABLE} --launcher-no-splash --launch)
+# Enable/Disable Slicer custom modules: To create a new module, use the SlicerExtensionWizard.
+set(Slicer_EXTENSION_SOURCE_DIRS
+  ${${APPLICATION_NAME}_SOURCE_DIR}/Modules/MatlabModules
+  ${${APPLICATION_NAME}_SOURCE_DIR}/Modules/Scripted/Home
```

Comment:
If I add another python module, could we add `${${APPLICATION_NAME}_SOURCE_DIR}/Modules/Scripted` here instead of `${${APPLICATION_NAME}_SOURCE_DIR}/Modules/Scripted/Home`?

Reply:
If think it make more sens to have all parameter in the top level CMakeLists, it is nice to have the big picture in one location. If you have more you could do something like this:

```
foreach(moduleName IN ITEMS Scripted1 Scripted2)
  list(APPEND Slicer_EXTENSION_SOURCE_DIRS
    ${${APPLICATION_NAME}_SOURCE_DIR}/Modules/Scripted/${moduleName}
    )
endif()
```